### PR TITLE
Fix query replayer crash

### PR DESCRIPTION
**Pull request by tanjialiang**
Issues with current replayer
1) Query replayer does not register hive connector factory.
2) PartitionedOutputReplayer holds non-existent reference on default consumer.

The PR fixes these 2 issues

### CODE REVIEW

**Review by duanmeng — APPROVED** [review]
@tanjialiang Nice fix! It seems that we must register connector factories in the client after #8871

File: velox/tool/trace/tests/PartitionedOutputReplayerTest.cpp [hunk 1]
```diff
@@ -76,10 +76,81 @@ class PartitionedOutputReplayerTest : public HiveConnectorTestBase {
...
+  auto consumerExecutor = std::make_unique<folly::CPUThreadPoolExecutor>(
+      numPartitions, std::make_shared<folly::NamedThreadFactory>("Consumer"));
+  consumeAllData(
+      bufferManager_,
+      originalTask->taskId(),
+      numPartitions,
+      executor_.get(),
+      consumerExecutor.get(),
+      [&](auto /* unused */, auto /* unused */) {});
+  ASSERT_NO_THROW(
```
Comment (xiaoxmeng) [inline velox/tool/trace/tests/PartitionedOutputReplayerTest.cpp:148]: VELOX_ASSERT_THROW

File: velox/tool/trace/tests/PartitionedOutputReplayerTest.cpp [hunk 2]
```diff
@@ -76,10 +76,81 @@ class PartitionedOutputReplayerTest : public HiveConnectorTestBase {
...
+       makeFlatVector<int32_t>(1'000, [](auto row) { return row; })});
+  const auto testDir = TempDirectoryPath::create();
+  const auto traceRoot = fmt::format("{}/{}", testDir->getPath(), "traceRoot");
+  auto originalTask = createPartitionedOutputTask(
+      {input},
+      {"key"},
+      {"key", "value"},
+      traceRoot,
+      fmt::format(
+          "local://test-partitioned-output-replayer-basic-{}",
```
Comment (xiaoxmeng) [inline velox/tool/trace/tests/PartitionedOutputReplayerTest.cpp:132]: "local://test-partitioned-output-replayer-basic-defaultConsumer"?

**Review by xiaoxmeng — COMMENTED** [review]
@tanjialiang LGTM % nits. Thanks!

**Review by xiaoxmeng — APPROVED** [review]

**Pull request closed without merging**